### PR TITLE
[Plots.Pie] Labels are removed when labelsEnabled(false)

### LINKED ISSUE #2483: [Plots.Pie] Setting labelsEnabled(false) after rendering does not erase labels
Repro: http://jsfiddle.net/xx94nr1t/

**Actual**: Labels continue to be shown
**Expected**: `labelsEnabled(false)` will hide labels if they were previously shown.

### PR DESCRIPTION
**Pull request by jtlan**
Also fixed:
* Updating data doesn't create multiple sets of labels.
* Updating data recomputes slices before rendering.

... both of which were found in the course of investigating the main bug.

Close #2483.

### CODE REVIEW

File: test/plots/piePlotTests.ts [hunk 1]
```diff
@@ -14,6 +14,119 @@ describe("Plots", () => {
...
+      assert.lengthOf(fourSlicePathStrings, 4, "one path per datum");
+
+      var twoSliceData = [
+        { value: 1 },
+        { value: 1 }
+      ];
+      dataset.data(twoSliceData);
+      var twoSlicePathStrings: String[] = [];
+      piePlot.content().selectAll("path").each(function() { twoSlicePathStrings.push(d3.select(this).attr("d")); });
+      assert.lengthOf(twoSliceData, 2, "one path per datum");
```
Comment (aicioara) [inline test/plots/piePlotTests.ts]: I guess you meant `twoSlicePathStrings`

Comment (jtlan) [inline test/plots/piePlotTests.ts]: Indeed I did...

File: test/plots/piePlotTests.ts [hunk 2]
```diff
@@ -14,6 +14,119 @@ describe("Plots", () => {
...
+        { value: 1 },
+        { value: 1 },
+        { value: 1 }
+      ];
+      var dataset = new Plottable.Dataset(fourSliceData);
+      piePlot.addDataset(dataset);
+      piePlot.renderTo(svg);
+      var fourSlicePathStrings: String[] = [];
+      piePlot.content().selectAll("path").each(function() { fourSlicePathStrings.push(d3.select(this).attr("d")); });
+      assert.lengthOf(fourSlicePathStrings, 4, "one path per datum");
```
Comment (aicioara) [inline test/plots/piePlotTests.ts]: I don't like the `4`
```typescript
assert.lengthOf(fourSlicePathStrings, fourSliceData.length, "one path per datum")
```

File: test/plots/piePlotTests.ts [hunk 3]
```diff
@@ -14,6 +14,119 @@ describe("Plots", () => {
...
+        piePlot.labelsEnabled(true);
+        var data1 = [
+          { value: 1 },
+          { value: 1 },
+          { value: 1 }
+        ];
+        var dataset = new Plottable.Dataset(data1);
+        piePlot.addDataset(dataset);
+        piePlot.renderTo(svg);
+        piePlot.labelsEnabled(false);
```
Comment (aicioara) [inline test/plots/piePlotTests.ts]: Assert that they actually existed before disabling them

File: test/plots/piePlotTests.ts (hunk 2 above)
Comment (jtlan) [inline test/plots/piePlotTests.ts]: Sounds good.

File: test/plots/piePlotTests.ts (hunk 3 above)
Comment (jtlan) [inline test/plots/piePlotTests.ts]: Also a good idea.

**Pull request merged**